stlab/concurrency/main_executor.hpp

```diff
@@ -104,7 +104,28 @@ struct main_executor_type {
 
 #elif STLAB_MAIN_EXECUTOR(EMSCRIPTEN)
 
-using main_executor_type = default_executor_type;
+struct main_scheduler_type {
```

Comment:
Did you consider adding a emscripten-specific unit test that verifies this code?

Reply:
I did - but the difficulty of implementation gave me pause. We'd need a way to have a JS run loop to test it. I think that's a sizeable enough addition to the testing architecture that we're better off tracking that work as part of #406 so we can get this CI support in.

Reply:
That said please feel free to push back if you think that should block merging this PR.

Reply:
I agree to track this one with #406 - we'll have to investigate how to code unit tests with the JS run loop.

Reply:
okay